static/app/views/issueList/editableTabTitle.tsx

```diff
@@ -0,0 +1,120 @@
+import {useCallback, useEffect, useRef, useState} from 'react';
+import styled from '@emotion/styled';
+
+import useKeyPress from 'sentry/utils/useKeyPress';
+import useOnClickOutside from 'sentry/utils/useOnClickOutside';
+
+function EditableTabTitle({
+  label,
+  onChange,
+  isEditing,
+  setIsEditing,
+  inputRef,
+}: {
+  isEditing: boolean;
+  label: string;
+  onChange: (newLabel: string) => void;
+  setIsEditing: (isEditing: boolean) => void;
+  inputRef?: React.RefObject<HTMLInputElement>;
+}) {
+  const [inputValue, setInputValue] = useState(label);
+
+  const wrapperRef = useRef<HTMLDivElement>(null);
+
+  const enter = useKeyPress('Enter', undefined, true);
+  const esc = useKeyPress('Escape', undefined, true);
+
+  const isEmpty = !inputValue.trim();
+
+  useOnClickOutside(wrapperRef, () => {
+    if (!isEditing) {
+      return;
+    }
+    if (isEmpty) {
+      setInputValue(label);
+      return;
+    }
+    if (inputValue !== label) {
+      onChange(inputValue);
+    }
+
+    setIsEditing(false);
+  });
+
+  const onEnter = useCallback(() => {
+    if (enter) {
+      if (isEmpty) {
+        setInputValue(label);
+        return;
+      }
+      if (inputValue !== label) {
+        onChange(inputValue);
+      }
+
+      setIsEditing(false);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [enter, inputValue, onChange]);
+
+  const onEsc = useCallback(() => {
+    if (esc) {
+      if (label !== inputValue) {
+        setInputValue(label);
+      }
+      if (isEditing) {
+        setIsEditing(false);
+      }
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [esc]);
+
+  useEffect(() => {
+    if (isEditing) {
+      onEnter();
+      onEsc();
+    }
+  }, [onEnter, onEsc, isEditing]);
+
+  useEffect(() => {
+    setTimeout(() => {
+      inputRef?.current?.focus();
```

Comment:
Can you use the `autoFocus` attribute (on the input) instead?

Reply:
This doesn't work for some reason :/

Reply:
Hmm it also seems like you don't need the `setTimeout` either. I wonder if something attempts to steal focus when clicking edit

Reply:
Oh interesting - the setTimeout solved a previous issue I had, but it sounds like that's no longer an issue. Same thing happened with the `onInputChange = {e => e.stopPropogation()}` - I deleted it after a few other changes, and then the issue just didn't appear anymore.